### ISSUE TITLE
Single checkbox attribute issue

### ISSUE THREAD

**Issue by qzminski** [isotope/core#1658]
If you create the attribute "checkbox menu" with a single option created via Options Manager, then you won't be able to check and save that field in the product settings. The reason is that the options manager creates the options array including the option database record ID and the option label:

```php
$GLOBALS['TL_DCA']['tl_iso_product']['fields']['my_field']['options'] = [
    123 => 'My option',
];
```

That is correct for the multiple checkboxes but the single one gets **always** value=1, see:

https://github.com/contao/core/blob/master/system/modules/core/widgets/CheckBox.php#L188

Thus when you attempt to save the record you will get ```Invalid input: 1``` error message because there is no option with key ```1```.

**Comment by aschempp** [comment]
I would say this is a Contao issue, can you think of a good solution?

**Comment by qzminski** [comment]
I would say "if there is no multiple flag, always use 1 as option value".

**Comment by aschempp** [comment]
What if the option has a surcharge? We would not find the associated record in the database to calculate a price.

**Comment by qzminski** [comment]
We could also not not hardcode that ```1``` value but take the only option ID instead. However that possibly could be a BC break. Contao single checkbox by default indicates the yes or no state, so in your case it maybe would be better to check value by the field name, e.g. ```if ($_POST['surcharge_42'])...``` as you know the record ID anyway.

**Comment by aschempp** [comment]
And we would need a custom form field for that, right?

**Comment by qzminski** [comment]
Pardon me, for what exactly?

**Comment by aschempp** [comment]
Because the attribute is only responsible for generating the DCA, but the input is actually handled in the regular Contao form field.

**Comment by qzminski** [comment]
Yes, then a custom form field is needed. Or as I said latter, it may be better to check just for the field name (e.g. ```surcharge_42```) and not the field value.

**Comment by aschempp** [comment]
Or maybe the issue should be fixed in Contao core? Because the option validation is actually wrong? @leofeyer

**Comment by leofeyer** [comment]
> We could also not not hardcode that `1` value but take the only option ID instead. However that possibly could be a BC break. 

What about this objection?

**Comment by aschempp** [comment]
I don't think that's a good idea, because it would result in different POST data. But the FormCheckbox class should probably not validate the options if it's not set to `multiple`?

**Comment by qzminski** [comment]
> But the FormCheckbox class should probably not validate the options if it's not set to ```multiple```?

Oh no, that's not the way to go. You want to read your record ID from an option value and have it not validated?

> @qzminski on the other hand I wonder if it makes sense to add checkbox options if there is only one? Or should we simply use the field/attribute name as option?

What does it change if we use field/attribute name as option?

I still opt for leaving things as they are and just check if the checkbox has been submitted ```if (\Input::post('checkbox_123')) { ... }```. The Isotope should be changed in a way that if there is only one option, make its value always ```1```. In a single checkbox use case it's always boolean state so no need to have a different option value there.

**Comment by aschempp** [comment]
> What does it change if we use field/attribute name as option?

As far as I remember `FormCheckbox` it does only use the option if there actually are options. So the validation would not fail if we do not define options (by simply not showing the options field if multiple checkbox is not set).

**Comment by qzminski** [comment]
Well actually ```FormCheckBox``` is irrelevant for this issue as the problem is in the backend only which is handled by the ```CheckBox``` class so...

> I would say "if there is no multiple flag, always use 1 as option value".

The frontend checkbox handling is a different story and not really related to this one. In the frontend it works without the hassle probably (see form generator for example).

**Comment by aschempp** [comment]
Ah, I didn't realize that. But that would be the same, right? If the attribute is not frontend (not variant attribute or customer editable), do not allow options when multiple is not checked.

I just wonder how to migrate existing options…

**Comment by qzminski** [comment]
The problem occurs only if there is a single option. If that's the case then option value should always become ```1```. There is no need to migrate the existing single options and they could have not been saved anyway because of the ```Invalid input: 1``` error (given value was > 1).

**Comment by aschempp** [comment]
Ok, so you agree we don't even need the options array in that case?

**Comment by qzminski** [comment]
Yup I think so. Remember that's only for backend ;-)

**Comment by aschempp** [comment]
@qzminski please prepare a PR to solve this.

**Comment by aschempp** [comment]
Closed in favor of #1684

**Comment by tsarma** [comment]
+1
